[C++][Release] Linux nightly verification jobs failing with Could NOT find LLVMAlt

### Describe the bug, including details regarding any error messages, version, and platform.

A cluster of Linux nightly verification jobs are failing to find LLVM using our FindLLVMAlt.cmake:

```
-- Could NOT find LLVM (missing: LLVM_DIR)
-- Could NOT find LLVM (missing: LLVM_DIR)
-- Could NOT find LLVM (missing: LLVM_DIR)
-- Could NOT find LLVM (missing: LLVM_DIR)
-- Could NOT find LLVM (missing: LLVM_DIR)
-- Could NOT find LLVM (missing: LLVM_DIR)
-- Could NOT find LLVM (missing: LLVM_DIR)
-- Could NOT find LLVM (missing: LLVM_DIR)
-- Could NOT find LLVM (missing: LLVM_DIR)
-- Could NOT find LLVM (missing: LLVM_DIR)
-- Could NOT find LLVM (missing: LLVM_DIR)
-- Could NOT find LLVM (missing: LLVM_DIR)
-- Could NOT find LLVM (missing: LLVM_DIR)
-- Could NOT find LLVM (missing: LLVM_DIR)
CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find LLVMAlt (missing: LLVM_PACKAGE_VERSION CLANG_EXECUTABLE
  LLVM_FOUND LLVM_LINK_EXECUTABLE)
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
  cmake_modules/FindLLVMAlt.cmake:125 (find_package_handle_standard_args)
  src/gandiva/CMakeLists.txt:30 (find_package)
```

(Excepted from https://github.com/ursacomputing/crossbow/actions/runs/12619675362/job/35164487847)

We'll need to resolve this before the 19.0.0 release.

### Component(s)

C++, Release

--- a/dev/release/setup-ubuntu.sh
+++ b/dev/release/setup-ubuntu.sh
@@ -42,7 +42,6 @@ apt-get install -y -q --no-install-recommends \
   bundler \
   clang \
   cmake \
-  llvm-dev \
   curl \
   git \
   gnupg \
@@ -51,6 +50,7 @@ apt-get install -y -q --no-install-recommends \
   libglib2.0-dev \
   libsqlite3-dev \
   libssl-dev \
+  llvm-dev \
   ninja-build \
   nlohmann-json3-dev \
   pkg-config \

--- a/dev/release/setup-ubuntu.sh
+++ b/dev/release/setup-ubuntu.sh
@@ -42,6 +42,7 @@ apt-get install -y -q --no-install-recommends \
   bundler \
   clang \
   cmake \
+  llvm-dev \
   curl \
   git \
   gnupg \

